inst/tests/tests.Rraw

```diff
@@ -10964,7 +10964,7 @@ test(1743.217, sapply(fread("a,b,c,d,e,f\na,b,c,d,e,f", colClasses = list(factor
 test(1743.218, sapply(fread("a,b,c,d,e,f\na,b,c,d,e,f", colClasses = list(factor = c(1, 2, 4), factor = 3), select = c(5, 4, 2, 3)), class), y = c(e = "character", d = "factor", b = "factor", c = "factor"))
 
 test(1743.22, fread("a,b,c\n1999/01/01,2,f", colClasses=list(Date=1L), drop="a"), data.table(b=2L, c="f"))
-test(1743.231, fread("a,b,c\n2,1,4i", colClasses=list(complex="c", integer=2L), drop="a"), data.table(b=1L, c="4i"),
+test(1743.231, fread("a,b,c\n2,1,4j", colClasses=list(complex="c", integer=2L), drop="a"), data.table(b=1L, c="4j"),
```

Comment:
This is actually a substantial improvement in r-devel -- previously it seems `as.complex("4i")` would fail to coerce which seems wrong. Better to try and coerce to complex something that's not actually complex, to ensure the warning continues.

cc original author @HughParsonage in case I'm missing something behind the intent of this test.

Reply:
I think Matt added that specific test, though I added the tests around it, which looks to have been a back and forth that observed but did not mention the behaviour of as.complex("4i"):

https://github.com/Rdatatable/data.table/pull/2545/commits/c6e2b72dc2ab2211b8333094345d3bc36a84175f#diff-e3243f3780ce7d303c3317f73945310bfc37e45d193568246246aca20e3270ae